src/main/java/org/spongepowered/api/command/args/parsing/SpaceSplitInputTokenizer.java

```diff
@@ -35,14 +35,18 @@
     private SpaceSplitInputTokenizer() {}
 
     @Override
-    public List<SingleArg> tokenize(String arguments, boolean lenient) throws ArgumentParseException {
-        List<SingleArg> ret = new ArrayList<>();
-        int lastIndex = 0;
-        int spaceIndex;
-        while ((spaceIndex = arguments.indexOf(" ")) != -1) {
-            ret.add(new SingleArg((arguments = arguments.substring(0, spaceIndex)), lastIndex, lastIndex + spaceIndex));
-            lastIndex += spaceIndex + 1;
+    public List<SingleArg> tokenize(String arguments, boolean lenient) {
```

Comment:
This has been dealt with in my previous PR (#1844) and so the changes SpaceSplitInputTokenizer and associated tests should be removed from this PR (with an exception, see your tests).

Reply:
I'll compare and rework as necessary to meet the test cases.